src/tike/communicators/mpi.py

```diff
@@ -0,0 +1,119 @@
+"""Define a MPI wrapper for inter-node communications.."""
+
+__author__ = "Xiaodong Yu"
+__copyright__ = "Copyright (c) 2021, UChicago Argonne, LLC."
+__docformat__ = 'restructuredtext en'
+
+from mpi4py import MPI
+
+import cupy as cp
+import numpy as np
+
+
+class MPIComm:
+    """A class for python MPI wrapper.
+
+    Many clusters do not support inter-node GPU-GPU
+    communications, so we first gather the data into
+    main memory then communicate them.
+
+    Attributes
+    ----------
+    gpu_count : int
+        The number of GPUs on each node.
```

Comment:
I think this attribute is unused and not needed? Only the `ThreadPool` needs to know about the GPU count?

Reply:
Removed.